src/Microsoft.AspNetCore.Analyzer.Testing/DiagnosticAnalyzerRunner.cs

```diff
@@ -65,7 +82,7 @@ protected Task<Diagnostic[]> GetDiagnosticsAsync(string[] sources, DiagnosticAna
                 var compilation = await project.GetCompilationAsync();
 
                 // Enable any additional diagnostics
-                var options = compilation.Options;
+                var options = ConfigureCompilationOptions(compilation.Options);
```

Comment:
By default this compiles to an exe and complains about not finding the Main method. I have it compiling to a library now with the option to modify it